_posts/en/newsletters/2020-09-23-newsletter.md

```diff
@@ -0,0 +1,264 @@
+    height or time.  However, the Blockstream post explains that this
+    also has the non-obvious implication that it's possible to create a
+    script that's unspendable because it requires the simultaneous use
+    of both heightlocks and timelocks.  For example, a payment to
+    following script can never be spent:
+
+       1 OP_CLTV OP_DROP 500000001 OP_CLTV
+
+    The first condition allows the spending transaction to be included
+    in block 1 or later, so any block after early January 2009, and the
+    second condition allows it to be included in any block after early
+    November 1985.  Both conditions are true today---but they can't both
+    be satisfied using a transaction's single nLockTime field.  The post
+    notes that the same problem can apply when a transaction has
+    multiple inputs each with their own script.  For example:
+
+        Input 0:
+          1 OP_CLTV
+
+        Input 1:
+          500000001 OP_CLTV
+
+    The problem also applies to relative timelocks and relative
+    heightlocks created using [BIP68][] sequence numbers and the
+    [BIP112][] `OP_CHECKSEQUENCEVERIFY` opcode.
+
+    The post notes that the [Miniscript][topic miniscript] compiler has been updated to
+    deal with the possible conflicts as best as possible.  It will
+    identify when one or more of the ways to satisfy a script contains
+    conflicting locks and will return a warning.  Because of the
+    conflict, it'll also not be able to provide full analysis of the
+    script.  Additionally, the compiler for the policy language will now
+    deliberately fail on policies that mix timelocks and heightlocks,
+    e.g.  `thresh(3,after(1),after(500000001),pk(A))`.  Note: as of this
+    writing, this change is only a [pending PR][rust-miniscript #121] to
+    the Rust version of the miniscript library and has not yet
+    propagated to the online live demos for [miniscript][miniscript demo] or [minsc][].
+
+## Changes to services and client software
+
+*In this monthly feature, we highlight interesting updates to Bitcoin
+wallets and services.*
+
+- **Swan supports sending to bech32 addresses:**
+  Swan, through their custodian Prime Trust, now [supports withdrawals][swan bech32 send]
+  to [bech32][topic bech32] addresses.
+
+- **Ledger adds manual coin selection support:**
+  Moving from a first-in, first-out UTXO selection model, Ledger now also
+  supports [manual coin selection][ledger coin control article] for either privacy
+  or fee-minimization benefits.
+
+- **Sparrow wallet announced:**
+  [Sparrow wallet][sparrow twitter thread] is a new desktop wallet supporting
+  single or multisignature addresses, PSBT, hardware wallets, and [coin
+  selection][topic coin selection].
+
+- **JoinMarket 0.7.0 adds BIP78, PSBT:**
+  [JoinMarket 0.7.0][joinmarket 0.7.0] includes support for [BIP78 payjoin][topic payjoin]
+  for payments. PSBT support was also added in order to facilitate the implementation.
+
```

Comment:
Since the last publishing of this section, one major thing happened with Wasabi Wallet.

With the recent v1.1.12 release [August 5th] https://github.com/zkSNACKs/WalletWasabi/releases/tag/v1.1.12, the Wasabi client is compatible with the v4 backend implementation of the ZeroLink CoinJoin protocol. On September 3rd, a coordination hardfork was enforced on the backend server, making it mandatory for users to upgrade to v1.1.12 if they wish to participate in a CoinJoin. Reason for the hardfork was a fix to a responsibly disclosed security vulnerability in the code. https://blog.wasabiwallet.io/responsible-disclosure-v4-hard-fork/

Sorry for providing this information so shortly before publishing.
However, maybe this is interesting for your readers, since you have mentioned Wasabi previously, and user action is required.

Reply:
@MaxHillebrand Thanks for the heads up, but going to pass on this item for the client/services section this month.